tree/ntuple/v7/src/RColumn.cxx

```diff
@@ -92,13 +92,19 @@ void ROOT::Experimental::Detail::RColumn::Flush()
 void ROOT::Experimental::Detail::RColumn::MapPage(const NTupleSize_t index)
 {
    fPageSource->ReleasePage(fReadPage);
+   // Set fReadPage to an empty page before populating it to prevent double destruction of the previously page in case
+   // the page population fails.
+   fReadPage = RPage();
```

Comment:
If we want to be extra careful, we could swap `fReadPage` with an empty page before calling `ReleasePage` - not sure if that can ever throw? Something like
```suggestion
   RPage currentPage = fReadPage;
   // Set fReadPage to an empty page before releaseing to prevent double destruction in case of an error.
   fReadPage = RPage();
   fPageSource->ReleasePage(currentPage);
```
Or should `ReleasePage` take care of that? (or maybe I'm just extra paranoid now...)

Reply:
The only way in which `ReleasePage` can fail is if `fReadPage` is not found in the in-memory page pool (e.g. because it already got released eearlier). This should already be prevented with the current proposed change and I don't really see how first swapping with an empty page will change anything since in the end we're still passing the same page to `ReleasePage` (unless I'm missing something?).

Reply:
Ok, fair enough, not yet really familiar with all the code paths in RNTuple. Yes, we would still pass the same `RPage` to `ReleasePage`.